src/index.ts

```
@@ -73,7 +82,19 @@ const processor$ = combineLatest(
         configManager.mqtt$,
     )
     .pipe(
-        switchMap(([ dobissConfig, mqttConfig ]) => {
+        switchMap(([ dobissConfig, mqttConfig ]): Observable<Config> => {
+            if (dobissConfig.host) {
+                return of({dobissConfig, mqttConfig});
+            } else {
+                return from(arp.getTable()).pipe(
+                    map((arpTabel) => {
+                        dobissConfig.host = arpTabel.find(x => x.vendor === 'Arm')?.ip ?? '192.168.0.10';

```

I'm not sure 192.168.0.10 is a good fallback. It'll be confusing if discovery ended up not working (eg: container in different network). Maybe we should throw an error and log something in this case ? That auto-discovery did not work and the hostname should be set manually or that they should verify the service is in the right network. Also, what if you have multiple entries that match vendor === 'Arm' ?

Also, it'd be nicer if this was all embedded in the `dobiss$` observable exposed by config.ts. Then there is no need to change the index.ts. In the end I'd like to move out as much logic from the index as possible.

---

I've removed the fallback. It was your default so i didn't want to mess things up for you. We should check the ports when we have multiple same vendors. I have a few pi's in my network but none report Arm as mac address. I went on a limb and we can change this later, i guess.

After some testing, i'm not sure if the arp table fetched by the library is correct. After an nmap, my result is correct. So that's a possible bug i will fix if it comes up again.